src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -2292,6 +2364,60 @@ Vector2i WindowImplX11::getPrimaryMonitorPosition()
     return monitorPosition;
 }
 
+void WindowImplX11::setItemDroppingEnabled(bool enabled)
+{
+    // In order for file dropping to be enabled, the XdndAware property must be set.
+    if (enabled)
+    {
+        Atom xdnd_version = 5;
+        XChangeProperty(m_display,
+                        m_window,
+                        getAtom("XdndAware"),
+                        XA_ATOM,
+                        32,
+                        PropModeReplace,
+                        reinterpret_cast<unsigned char*>(&xdnd_version),
+                        true);
+    }
+    else
+    {
+        XDeleteProperty(m_display, m_window, getAtom("XdndAware"));
+    }
+}
+
+sf::String WindowImplX11::getWindowStringProperty(Display* dpy, Window w, Atom p)
+{
+    // Taken from https://www.uninformativ.de/blog/postings/2017-04-02/0/POSTING-en.html
+
+    Atom           da, incr, type;
+    int            di;
+    unsigned long  size, dul;
+    unsigned char* prop_ret = NULL;
```

Comment:
Please declare variables as late as possible -- only once you need them. If possible, initialize them immediately during initialization.

Declaring everything at the beginning of the function is relatively outdated C practice, either mandated by old compilers or a way to measure stack memory usage.

Reply:
I took this code from https://www.uninformativ.de/blog/postings/2017-04-02/0/POSTING-en.html, and i didn't test it very thoroughly, I just wanted to make a quick draft PR so we could figure out how to get it to work on multiple platforms, and then I would go and polish it up before merging it into master.

Reply:
I'd also like to see Bromeon's comments be addressed even if this is just a draft PR.

Reply:
Ok, I've completely changed how the selection is converted. I modded the conversion code from https://github.com/SFML/SFML/blob/master/src/SFML/Window/Unix/ClipboardImpl.cpp to work with mine.